accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,416 @@
+- Feature Name: Container as a Service Platform integration with Uyuni/SUSE Manager
+- Start Date: 2019-08-05
+- RFC PR: https://github.com/uyuni-project/uyuni-rfc/pull/10
+
+# Summary
+[summary]: #summary
+
+This RFC describes how Uyuni/SUSE Manager can be leveraged to perform cross-product integration with a cluster solution like Container as a Service Platform.
+
+# Motivation
+[motivation]: #motivation
+
+<!-- - Why are we doing this?
+- What use cases does it support?
+- What is the expected outcome?
+
+Describe the problem you are trying to solve, and its constraints, without coupling them too closely to the solution you have in mind. If this RFC is not accepted, the motivation can be used to develop alternative solutions. -->
+
+Uyuni/SUSE Manager is the systems management tool and must offer to the end-user the possibility to deploy and manage other SUSE products like CaaS Platform.
+
+We will be focusing first on the following general scenarios:
+
+1. Deploy a cluster using Uyuni/SUSE Manager
+2. Manage a cluster using Uyuni/SUSE Manager
+
+The two subsequent CaaS Platform-specific sections will follow.
+
+# Detailed design
+[design]: #detailed-design
+
+<!-- This is the bulk of the RFC. Explain the design in enough detail for somebody familiar with the product to understand, and for somebody familiar with the internals to implement.
+
+This section should cover architecture aspects and the rationale behind disruptive technical decisions (when applicable), as well as corner-cases and warnings. Whenever the new feature creates new user interactions, this section should include examples of how the feature will be used. -->
+
+## Generic design: integrate Uyuni/SUSE Manager with a cluster provider
+
+### Deploy a cluster using Uyuni/SUSE Manager
+
+Under Uyuni/SUSE Manager main menu, a new menu item "Clusters" should be introduced.
+All deployable and manageable cluster products (e.g. CaaSP, SOC, SES) will be in the sub-menu.
```

Comment:
Now that we added the `It is intended that Uyuni/SUSE Manager will not deal and will not know with the application deployed on top of CaaS Platform.` note, I don't see how this paragraph fits (`and manageable cluster products will be inthe sub-menu`).

Reply:
We are talking about two different things here: one is the application that runs on top of CaaSP cluster. This is specified in the CaaSP section.
The other is the type of cluster managed by SUSE Manager: is it a CaaSP cluster? a SOC cluster? This is specified in the generic cluster handling section.

Does this clarify?

Reply:
+1, thank you.

Reply:
Still I think the type of cluster in the submenu is a bit weird. Eg. for the `Images` section we support various types of images in the model, but we don't separate them like that in the menu.

![Screenshot from 2019-09-17 14-12-34](https://user-images.githubusercontent.com/15332/65040683-3f5cba80-d955-11e9-9fb7-72b806490cdc.png)

Once we add other cluster types, how this will grow? multiple instances of the same cluster type?

Reply:
> Still I think the type of cluster in the submenu is a bit weird. Eg. for the `Images` section we support various types of images in the model, but we don't separate them like that in the menu.
> 
> ![Screenshot from 2019-09-17 14-12-34](https://user-images.githubusercontent.com/15332/65040683-3f5cba80-d955-11e9-9fb7-72b806490cdc.png)
> 
> Once we add other cluster types, how this will grow? multiple instances of the same cluster type?

I understand your point. But it is also true that we already have a second-level menu for distinguishing between system types: `Systems -> Systems List -> {Physical, Virtual, Inactive, ...}`

In the initial idea, in the Cluster menu, we list all registered clusters. The second-level menu would comprise all the operations that can be done specifically to that cluster type. This paradigm is a mix between the `Systems` paradigm above and the `Images` you pointed out (or all other sections that have sub-actions as a second-level menu).

Do you think that does this fit into the "UI and coherence" point of view?